uaparser/parser.go

```diff
@@ -152,11 +153,13 @@ const (
 	cDefaultMissesTreshold = 500000
 	cDefaultMatchIdxNotOk  = 20
 	cDefaultSortOption     = false
+	cDefaultCacheSize      = 1024
 )
 
 var (
-	missesTreshold = uint64(500000)
-	matchIdxNotOk  = 20
+	missesTreshold      = uint64(500000)
+	matchIdxNotOk       = 20
+	parserCacheSize     = 1024
```

Comment:
```suggestion
```
per my comments below - module level variables add the possibility of racing between multiple threads, so it's better to keep these passed as parameters than in module scope.  Since you are not touching the preexisting variables I'm fine if you leave those two as is (potentially we'll refactor later) but I think it doesn't make sense to pile on that tech debt.